File: cpp_solutions/1461. Check If a String Contains All Binary Codes of Size K.cpp
class Solution {
public:
    bool hasAllCodes(string s, int k) {
        unordered_set<string> temp;
        
        if (k > s.size()){
            return false;
        }
        for(int x=0;x<s.length()-k+1;x++){
            temp.insert(s.substr(x,k));
        }
        
        return temp.size() == pow(2,k);
        
    }
};

File: python3_solution_set1/1461. Check If a String Contains All Binary Codes of Size K.py
class Solution:
    def hasAllCodes(self, s: str, k: int) -> bool:
        
        temp = set()
        
        for x in range(len(s)-k+1):
            temp.add(s[x:x+k])
            
        return len(temp) == 2 ** k
        
